### PR TITLE
Add support for entrypoint annotation

### PR DESCRIPTION
**Pull request by alexarchambault**

### CODE REVIEW

File: modules/directives/src/main/scala/scala/build/preprocessing/directives/Entrypoints.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import scala.build.directives.*
+import scala.build.errors.BuildException
+import scala.build.options.{BuildOptions, PostBuildOptions}
+import scala.cli.commands.SpecificationLevel
+
+@DirectiveExamples("//> using entrypoints")
+@DirectiveUsage("//> using entrypoints", "`//> using entrypoints`")
+@DirectiveDescription("Enable entrypoint annotation support")
+@DirectiveLevel(SpecificationLevel.EXPERIMENTAL)
+final case class Entrypoints(
```
Comment (tgodzik) [inline modules/directives/src/main/scala/scala/build/preprocessing/directives/Entrypoints.scala:12]: Sorry for the drive-by comment, but couldn't this be added just by using `//> using lib "com.github.alexarchambault::case-app-entrypoint-annotation:0.1.0-SNAPSHOT"` ? Is a separate directive needed?

File: modules/integration/src/test/scala/scala/cli/integration/RunEntrypointsTests.scala [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  val extraArgs = Seq(
+    "--scala=3.3.0-RC1-bin-20230120-d6cc101-NIGHTLY",
+    "--server=false",
+    "-r",
+    "sonatype:snapshots"
+  )
+
+  test("simple") {
+    val inputs = TestInputs(
+      os.rel / "Hello.scala" ->
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/RunEntrypointsTests.scala:16]: would `@entrypoint` also work with `.sc` files?

File: modules/directives/src/main/scala/scala/build/preprocessing/directives/Entrypoints.scala (hunk 1 above)
Comment (alexarchambault) [reply modules/directives/src/main/scala/scala/build/preprocessing/directives/Entrypoints.scala:12]: Having the using directive should make the completion stuff try to actually compile the project, and provide completions for the entrypoints' options. Ideally, it should also import the annotation itself.

File: modules/integration/src/test/scala/scala/cli/integration/RunEntrypointsTests.scala (hunk 2 above)
Comment (alexarchambault) [reply modules/integration/src/test/scala/scala/cli/integration/RunEntrypointsTests.scala:16]: It ought to, but it currently runs into TASTy version issues (that I need to look into)

**Comment by Gedochao** [comment]
Closing this for now.
It seems we wouldn't be able to merge it until macro annotations become stable in Scala 3, which doesn't have a date just yet.

**Comment by Gedochao** [comment]
Closing this for now.
It seems we wouldn't be able to merge it until macro annotations become stable in Scala 3, which doesn't have a date just yet.
